lib/utils/queryable.js

```diff
@@ -0,0 +1,260 @@
+      res.push(k)
+    else {
+      // splits the dot-sep property names and add them to the list of keys
+      for (const splitKey of k.split('.'))
+        /* eslint-disable-next-line no-new-wrappers */
+        res.push(new String(splitKey))
+    }
+  }
+
+  // returns an ordered list of strings in which each entry
+  // represents a key in an object defined by the previous entry
+  return res
+}
+
+const getter = ({ data, key }) => {
+  // keys are a list in which each entry represents the name of
+  // a property that should be walked through the object in order to
+  // return the final found value
+  const keys = parseKeys(key)
+  let _data = data
+  let label = ''
+
+  for (const k of keys) {
+    // extra logic to take into account printing array, along with its
+    // special syntax in which using a dot-sep property name after an
+    // arry will expand it's results, e.g:
+    // arr.name -> arr[0].name=value, arr[1].name=value, ...
+    const maybeIndex = Number(k)
+    if (Array.isArray(_data) && !Number.isInteger(maybeIndex)) {
+      _data = _data.reduce((acc, i, index) => {
+        acc[`${label}[${index}].${k}`] = i[k]
+        return acc
+      }, {})
+      return _data
+    } else {
+      // if can't find any more values, it means it's just over
+      // and there's nothing to return
+      if (!_data[k])
+        return undefined
+
+      // otherwise sets the next value
+      _data = _data[k]
+    }
+
+    label += k
+  }
+
+  // these are some legacy expectations from
+  // the old API consumed by lib/view.js
+  if (Array.isArray(_data) && _data.length <= 1)
+    _data = _data[0]
+
+  return {
+    [key]: _data,
+  }
+}
+
+const setter = ({ data, key, value, force }) => {
+  // setter goes to recursively transform the provided data obj,
+  // setting properties from the list of parsed keys, e.g:
+  // ['foo', 'bar', 'baz'] -> { foo: { bar: { baz:  {} } }
+  const keys = parseKeys(key)
+  const setKeys = (_data, _key) => {
+    // handles array indexes, making sure the new array is created if
+    // missing and properly casting the index to a number
+    const maybeIndex = Number(_key)
+    if (!Number.isNaN(maybeIndex)) {
+      _key = maybeIndex
+      if (!Object.keys(_data).length)
+        _data = []
+    }
+
+    // retrieves the next data object to recursively iterate on,
+    // throws if trying to override a literal value or add props to an array
+    const next = () => {
+      const haveContents =
+        !force &&
+        _data[_key] != null &&
+        value !== _delete
+      const shouldNotOverrideLiteralValue =
+        !(typeof _data[_key] === 'object')
+      // if the next obj to recurse is an array and the next key to be
+      // appended to the resulting obj is not an array index, then it
+      // should throw since we can't append arbitrary props to arrays
+      const shouldNotAddPropsToArrays =
+        Array.isArray(_data[_key]) &&
+        Number.isNaN(Number(keys[0]))
```

Comment:
```suggestion
        isNaN(keys[0])
```
altho your way's more explicit

Reply:
yeah, let's go with the more explicit alternative 😊